contrib/cirrus/integration_test.sh

```diff
@@ -23,6 +23,10 @@ case "${OS_RELEASE_ID}-${OS_RELEASE_VER}" in
     fedora-28) ;&
     centos-7) ;&
     rhel-7)
+        make build-all-new-commits BASE_BRANCH=$CIRRUS_BASE_BRANCH
+        # Be sure we don't leave any file from the rebase
+        git reset --hard
+        git clean -fdx
```

Comment:
If it makes sense, this could be run in parallel with the 'unit' task, then the testing & image-building tasks made dependent.  We already eat the ~1min VM-creation time in all cases, so adding another VM in parallel only speeds things up.  The other advantage is, it creates a separate/distinct set of output from the unittests, so spotting this specific failure is faster.

OTOH, what you have here is fine also, and adding a minute or so to unittesting is a small price given the simplicity here.  So I'm fine either way.

Reply:
Ohh, actually hang on...

Reply:
...I think the ``git reset --hard`` is going to cause problems later on.  After the build, better reset the repo. to ``$CIRRUS_CHANGE_IN_REPO`` (I think that's the right var.).

Reply:
Otherwise the integration tests will run against `origin/master` instead of testing the code from the PR.